### PR TITLE
boost 1.78.0

### PR DESCRIPTION
**Pull request by SMillerDev**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by iMichka** [comment]
I think we will have to introduce a `boost@1.76` fomula, at least for `libbitcoin`, see https://github.com/libbitcoin/libbitcoin-system/issues/1234

**Comment by SMillerDev** [comment]
@Homebrew/ops any idea why the linux build never finishes?

**Comment by iMichka** [comment]
```
Full audit gnuradio --git --skip-style output
  gnuradio:
    * Libraries were compiled with a flat namespace.
      This can cause linker errors due to name collisions, and
      is often due to a bug in detecting the macOS version.
        /usr/local/Cellar/gnuradio/3.9.3.0_2/lib/libgnuradio-soapy.3.9.3.0.dylib
  Error: 1 problem in 1 formula detected
```

Let's fix this first, this should give us a green build for the macOS side. And then let's re-run this.

I don't know what happened with the Linux runner.

A rebase will also be needed for `supertux`.

**Comment by SMillerDev** [comment]
Can't find any definitions of `-flat_namespace` in the gnuradio source. Maybe CMake added this somehow?

**Comment by carlocab** [comment]
Probably comes from a dependency. But it's a CMake build, so this isn't because of the libtool bug. I'm fine with adding this to the allowlist.

**Comment by iMichka** [comment]
I would like to investigate this a little bit more. But this does not prevent us to ship the boost update.
We need to re-run the CI here though, as the Linux build did not work at all.

**Comment by iMichka** [comment]
I think I found a solution: #92726

**Comment by carlocab** [comment]
`openimageio` failed on Linux because it build Python3.10 bindings for some reason. The formula expects Python3.9.

**Comment by carlocab** [comment]
`openimageio` fixed in #92931.

**Comment by carlocab** [comment]
`cppzmq` failed because of a conflicting `zmq.hpp` in `gnuradio` (which possibly vendors a `cppzmq`).

`libmwaw` failed because of a failure to download the test resource.

`libswiften` needs a `revision` bump. It may not build against the newest `boost`.

**Comment by iMichka** [comment]
I have a PR for `gnuradio`: #93003

`libmwaw` looks transient: it's a file hosted on GitHub and the file is present. It was a 503 error, and I have seen a few of these over the last days.

**Comment by iMichka** [comment]
Fix for `rdkit`: #93185

**Comment by iMichka** [comment]
Fix for `libswiften`: #93296

**Comment by carlocab** [comment]
Pushed a fix for `cryptominisat` which failed on ARM.

**Comment by iMichka** [comment]
Looks that was not enough to fix it:
```
2022-01-18T19:23:53.7944560Z /opt/homebrew/Library/Homebrew/shims/mac/super/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -DLIBRARY_VERSION=\"5.8.0\" -I/opt/homebrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/pycryptosat/src/pycryptosat.cpp -o build/temp.macosx-12-arm64-3.9/tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/pycryptosat/src/pycryptosat.o -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -I/tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0 -I/tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src
2022-01-18T19:23:53.7945990Z In file included from /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/pycryptosat/src/pycryptosat.cpp:32:
2022-01-18T19:23:53.7946550Z In file included from /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/cryptominisat.h:35:
2022-01-18T19:23:53.7947160Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:33:1: error: unknown type name 'constexpr'
2022-01-18T19:23:53.7947540Z constexpr uint32_t var_Undef(0xffffffffU >> 4);
2022-01-18T19:23:53.7947730Z ^
2022-01-18T19:23:53.7948180Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:33:19: error: expected ';' after top level declarator
2022-01-18T19:23:53.7948570Z constexpr uint32_t var_Undef(0xffffffffU >> 4);
2022-01-18T19:23:53.7948760Z                   ^
2022-01-18T19:23:53.7948930Z                   ;
2022-01-18T19:23:53.7949370Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:40:5: error: unknown type name 'uint32_t'
2022-01-18T19:23:53.7949710Z     uint32_t x;
2022-01-18T19:23:53.7949870Z     ^
2022-01-18T19:23:53.7950300Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:41:5: error: unknown type name 'constexpr'
2022-01-18T19:23:53.7950670Z     constexpr explicit Lit(uint32_t i) : x(i) { }
2022-01-18T19:23:53.7950870Z     ^
2022-01-18T19:23:53.7951340Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:41:28: error: unknown type name 'uint32_t'
2022-01-18T19:23:53.7951700Z     constexpr explicit Lit(uint32_t i) : x(i) { }
2022-01-18T19:23:53.7951900Z                            ^
2022-01-18T19:23:53.7952370Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:41:24: error: constructor cannot have a return type
2022-01-18T19:23:53.7952750Z     constexpr explicit Lit(uint32_t i) : x(i) { }
2022-01-18T19:23:53.7952950Z                        ^~~
2022-01-18T19:23:53.7953400Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:43:5: error: unknown type name 'constexpr'
2022-01-18T19:23:53.7956950Z     constexpr Lit() : x(var_Undef<<1) {}   // (lit_Undef)
2022-01-18T19:23:53.7957200Z     ^
2022-01-18T19:23:53.7957730Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:43:15: error: constructor cannot have a return type
2022-01-18T19:23:53.7958110Z     constexpr Lit() : x(var_Undef<<1) {}   // (lit_Undef)
2022-01-18T19:23:53.7958310Z               ^~~
2022-01-18T19:23:53.7958760Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:44:5: error: unknown type name 'constexpr'
2022-01-18T19:23:53.7959150Z     constexpr explicit Lit(uint32_t var, bool is_inverted) :
2022-01-18T19:23:53.7959370Z     ^
2022-01-18T19:23:53.7959820Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:44:28: error: unknown type name 'uint32_t'
2022-01-18T19:23:53.7960200Z     constexpr explicit Lit(uint32_t var, bool is_inverted) :
2022-01-18T19:23:53.7960410Z                            ^
2022-01-18T19:23:53.7960890Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:44:24: error: constructor cannot have a return type
2022-01-18T19:23:53.7961350Z     constexpr explicit Lit(uint32_t var, bool is_inverted) :
2022-01-18T19:23:53.7961570Z                        ^~~
2022-01-18T19:23:53.7962020Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:48:5: error: unknown type name 'constexpr'
2022-01-18T19:23:53.7962460Z     constexpr const uint32_t& toInt() const { // Guarantees small, positive integers suitable for array indexing.
2022-01-18T19:23:53.7962730Z     ^
2022-01-18T19:23:53.7963180Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:48:29: error: expected ';' at end of declaration list
2022-01-18T19:23:53.7963620Z     constexpr const uint32_t& toInt() const { // Guarantees small, positive integers suitable for array indexing.
2022-01-18T19:23:53.7963890Z                             ^
2022-01-18T19:23:53.7964060Z                             ;
2022-01-18T19:23:53.7964530Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:94:28: error: use of undeclared identifier 'var_Undef'
2022-01-18T19:23:53.7964940Z static const Lit lit_Undef(var_Undef, false);  // Useful special constants.
2022-01-18T19:23:53.7965180Z                            ^
2022-01-18T19:23:53.7965690Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:95:28: error: use of undeclared identifier 'var_Undef'; did you mean 'lit_Undef'?
2022-01-18T19:23:53.7966090Z static const Lit lit_Error(var_Undef, true );  //
2022-01-18T19:23:53.7966290Z                            ^~~~~~~~~
2022-01-18T19:23:53.7966460Z                            lit_Undef
2022-01-18T19:23:53.7966920Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:94:18: note: 'lit_Undef' declared here
2022-01-18T19:23:53.7993110Z static const Lit lit_Undef(var_Undef, false);  // Useful special constants.
2022-01-18T19:23:53.7993500Z                  ^
2022-01-18T19:23:53.7997820Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:95:18: error: no matching constructor for initialization of 'const CMSat::Lit'
2022-01-18T19:23:53.7998240Z static const Lit lit_Error(var_Undef, true );  //
2022-01-18T19:23:53.7998460Z                  ^         ~~~~~~~~~~~~~~~
2022-01-18T19:23:53.7999050Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:38:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 2 were provided
2022-01-18T19:23:53.7999460Z class Lit
2022-01-18T19:23:53.7999620Z       ^
2022-01-18T19:23:53.8000200Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:38:7: note: candidate constructor (the implicit default constructor) not viable: requires 0 arguments, but 2 were provided
2022-01-18T19:23:53.8001050Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:99:13: error: invalid operands to binary expression ('const CMSat::Lit' and 'const CMSat::Lit')
2022-01-18T19:23:53.8001450Z     if (lit == lit_Undef) {
2022-01-18T19:23:53.8001620Z         ~~~ ^  ~~~~~~~~~
2022-01-18T19:23:53.8002150Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:102:20: error: no member named 'sign' in 'CMSat::Lit'
2022-01-18T19:23:53.8002590Z         os << (lit.sign() ? "-" : "") << (lit.var() + 1);
2022-01-18T19:23:53.8002790Z                ~~~ ^
2022-01-18T19:23:53.8003330Z /tmp/cryptominisat-20220118-51428-1paierq/cryptominisat-5.8.0/build/cmsat5-src/cryptominisat5/solvertypesmini.h:102:47: error: no member named 'var' in 'CMSat::Lit'
2022-01-18T19:23:53.8003910Z         os << (lit.sign() ? "-" : "") << (lit.var() + 1);
2022-01-18T19:23:53.8004130Z                                           ~~~ ^
2022-01-18T19:23:53.8004450Z fatal error: too many errors emitted, stopping now [-ferror-limit=]
2022-01-18T19:23:53.8004690Z 20 errors generated.
```

**Comment by carlocab** [comment]
Huh, that's annoying. Though I guess not too surprising, given that I just realised their `CMakeLists.txt` already sets `CMAKE_CXX_STANDARD` to `11`. Weird

**Comment by carlocab** [comment]
Yea, `cryptominisat` refuses to build with the latest `boost`. Will try switching it to the older one.

**Review by iMichka — DISMISSED** [review]

**Review by iMichka — CHANGES_REQUESTED** [review]
I was able to build `cryptominista` with `"-DENABLE_PYTHON_INTERFACE=FALSE"`, because the error happens in the `python/src` part. `-std=gnu++11` is missing in that specific compilation step. I'm looking into how to add it.

**Comment by carlocab** [comment]
> I was able to build `cryptominista` with `"-DENABLE_PYTHON_INTERFACE=FALSE"`, because the error happens in the `python/src` part. `-std=gnu++11` is missing in that specific compilation step. I'm looking into how to add it.

Ugly workaround:
```ruby
ENV["CXX"] = "#{ENV.cxx} -std=gnu++11"
```

File: Formula/cryptominisat.rb
Comment (SMillerDev) [inline Formula/cryptominisat.rb:38]:
```suggestion
                    "-DCMAKE_CXX_COMPILER=#{ENV.cxx}",
```
The build complains about this not being a path.

Comment (carlocab) [reply Formula/cryptominisat.rb:38]: Sigh, CMake is so annoying

**Comment by carlocab** [comment]
I have a patch for `cryptominisat` which I think should fix things: https://github.com/carlocab/cryptominisat/commit/7f7176b1fe8bf05c7d1031a5ecdb0a1a55ceb810

It'll require adding
```ruby
ENV.append_to_cflags "-std=gnu++11"
```
to the formula.

I want to test it before sending it upstream.

**Comment by iMichka** [comment]
I found a workaround that works:
`inreplace "python/setup.py.in", "cconf +", "cconf + ['-std=gnu++11'] +"`
 
The cconf variable comes from the PY_C_CONFIG variable, which are the CFLAGS provided by the Python config. Python's cflags do not provide `-std=gnu++11`.

We can't upstream the fix like this though.

**Comment by carlocab** [comment]
My patch is probably upstreamable. It mimics code they already have elsewhere: https://github.com/msoos/cryptominisat/blob/02f53d1fc045fdba53671306964d3d094feb949e/python/CMakeLists.txt#L85

**Comment by SMillerDev** [comment]
@carlocab anything I can do to help here?

**Comment by iMichka** [comment]
I tested the patch. The issue is that we would be adding `-std=gnu++11` to CFLAGS, which then breaks some cmake checks right at the beginning.
There is some more stuff going on with `PY_C_CONFIG`. I don't like the fact that C and CXX flags get mixed up here: it's c++ code that gets compiled, and there is no clear separation between C and CXX flags, so it's hard to inject c++ only flags.
Also, there seem to be some more code in setup.py.in that already tries to hack around flags provided by cmake, in the `_init_posix` function. This might need more investigation and a better patch.

**Comment by carlocab** [comment]
Ah yes I see. I suggest we go with @iMichka's `inreplace` for now to unblock this, but we should report the issue upstream so that it can be fixed.

**Comment by carlocab** [comment]
Pushed a fix for `standardese`. It probably should get a better one, but that will take a bit of time to work out.

**Comment by SMillerDev** [comment]
RDKit is downloading some libraries, so we should add `catch2` as a dependency in the next run.

Meanwhile this is a bigger issue since downloading the file does give the same hash on my machine:
```
  -- Could NOT find InChI in system locations (missing: INCHI_LIBRARY INCHI_INCLUDE_DIR) 
  Downloading https://www.inchi-trust.org/download/106/INCHI-1-SRC.zip...
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
  
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  100   299  100   299    0     0    783      0 --:--:-- --:--:-- --:--:--   788
  CMake Error at Code/cmake/Modules/RDKitUtils.cmake:235 (MESSAGE):
    The md5 checksum for
    /tmp/rdkit-20220215-36429-sp20jm/rdkit-Release_2021_09_4/External/INCHI-API/INCHI-1-SRC.zip
    is incorrect; expected: f2efa0c58cef32915686c04d7055b4e9, found:
    66b56e5161f020eb41a31adcb0ed2c0e
```

**Comment by carlocab** [comment]
The issue is unrelated to the `boost` bump, we could/should probably try to fix it separately so we can confirm the fix before rerunning this

**Comment by iMichka** [comment]
Testing `rdkit` here: #95248

**Comment by iMichka** [comment]
I've opened an issue for `cryptominisat` to notify upstream of our workaround: https://github.com/msoos/cryptominisat/issues/671

**Comment by iMichka** [comment]
I've opened a PR to test the `standardese` fix separately: #95254

**Review by iMichka — DISMISSED** [review]

**Comment by carlocab** [comment]
ARM builds are 🟢 🎉

**Comment by carlocab** [comment]
Opened #95354 to fix the `vowpal-wabbit` failure.

`libtorrent-rasterbar` has been fixed on the `RC_2_0` branch. We just need to track down the relevant patches.

**Comment by carlocab** [comment]
`sfcgal` fails on Linux with
```
  /home/linuxbrew/.linuxbrew/include/CGAL/Intersections_2/Segment_2_Segment_2.h:45:37: error: array must be initialized with a brace-enclosed initializer
     std::array<int, 2> pt_ids = {-1,-1};
                                       ^
```
No idea what's causing it. The error doesn't look like it should OS-specific, but this didn't fail on ARM.

**Review by carlocab — DISMISSED** [review]

**Comment by carlocab** [comment]
`gnuradio` needs a rev bump. A little conflicted about it. Rev bumping `gnuradio` will add another full day of CI here. This will likely need another rebase in the next week or so, since some of these formulae will be bumped by then, so I'm inclined to rebuild `gnuradio` separately.

On the other hand, we also risk `gnuradio` not building with Boost 1.78, which would be pretty bad since it has a significant number of users (~1000 installs per month).

**Comment by carlocab** [comment]
I think this has been delayed long enough -- I'll open a PR for gnuradio separately. There are also a handful of other PRs waiting for the `CI-long-timeout` label.

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
`collada-dom` probably just needs `uses_from_macos "libxml2"` and `lgogdownloader` needs `uses_from_macos "curl"`. `thrift` might need `uses_from_macos "zlib"`.

`standardese` might need a newer GCC. Same for `uhd`.

**Comment by carlocab** [comment]
`gnuradio` rebuilt in #95443.

**Pull request closed without merging**
